redbot/core/utils/views.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+import discord
+
+from discord.ext.commands import BadArgument
+from typing import TYPE_CHECKING, List, Dict, Union, Optional
+from redbot.core.commands.converter import get_dict_converter
+from redbot.core.i18n import Translator
+
+if TYPE_CHECKING:
+    from redbot.core.bot import Red
+
+_ = Translator("UtilsViews", __file__)
+
+
+class SetApiModal(discord.ui.Modal):
+    def __init__(
+        self,
+        bot: Red,
+        default_service: Optional[str] = None,
+        default_keys: Optional[Dict[str, str]] = None,
+    ):
+        self.bot: Red = bot
+        self.default_service = default_service
+        self.default_keys: List[str] = []
+        if default_keys is not None:
+            self.default_keys = list(default_keys.keys())
+        self.default_keys_fmt = self._format_keys(default_keys)
+
+        _placeholder_service = "service"
+        if self.default_service is not None:
+            _placeholder_service = self.default_service
+        _placeholder_token = "client_id YOUR_CLIENT_ID"
```

Comment:
Maybe client_secret can go here too? Clearing up any confusion of there being only one modal to fill in.